File: ruby/052_DiceProduct.rb
# 052 - Dice Product（★3）
# 因数分解

MOD = 10 ** 9 + 7
N = gets.to_i
A = Array.new(N) { gets.split.map(&:to_i) }
puts A.reduce(1) {|sum, arr| sum * arr.sum % MOD }

File: kotlin/052_DiceProduct.kt
/**
 * 052 - Dice Product（★3）
 * 因数分解
 */

fun main() {
  val M = 1_000_000_007L
  val N = readLine()!!.toInt()
  val A = List(N) { readLine()!!.split(" ").map(String::toLong) }
  val ans = A.fold(1L) { acc, a -> acc * a.sum() % M }
  println(ans)
}
